scripts/ts_error_checker.js

```diff
@@ -0,0 +1,199 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+/* eslint no-restricted-syntax: 0 */
+const { spawnSync } = require('child_process');
+const fs = require('fs');
+const path = require('path');
+
+// ANSI color codes for console output
+const colors = {
+  reset: '\x1b[0m',
+  red: '\x1b[31m',
+  green: '\x1b[32m',
+  yellow: '\x1b[33m',
+  blue: '\x1b[34m',
+  magenta: '\x1b[35m',
+  cyan: '\x1b[36m',
+  white: '\x1b[37m',
+  bold: '\x1b[1m',
+};
+
+const BASELINE_FILE = path.join(process.cwd(), 'ts_error_baseline.json');
+const TSC_BIN = path.join(process.cwd(), 'node_modules/.bin/tsc');
+const TSCONFIG_PATH = path.join(process.cwd(), 'tsconfig.check.json');
+
+console.log(
+  `${colors.cyan}━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━${colors.reset}`
+);
+console.log(`${colors.bold}TypeScript Error Checker${colors.reset}`);
+console.log(
+  `${colors.cyan}━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━${colors.reset}`
+);
+console.log(
+  `\n${colors.cyan}▶ ${colors.reset}Running TypeScript compiler to check for new errors...\n`
+);
+
+// Load baseline errors
+function loadBaseline() {
+  try {
+    if (fs.existsSync(BASELINE_FILE)) {
+      const content = fs.readFileSync(BASELINE_FILE, 'utf8');
+      const baseline = JSON.parse(content);
+      console.log(`${colors.blue}ℹ ${colors.reset}Loaded baseline from ${BASELINE_FILE}`);
+      console.log(
+        `${colors.blue}ℹ ${colors.reset}Baseline contains ${colors.bold}${baseline.errors.length}${colors.reset} known errors`
+      );
+      return baseline.errors;
+    } else {
+      console.warn(`${colors.yellow}⚠ ${colors.reset}Baseline file not found at ${BASELINE_FILE}.`);
+      console.warn(`${colors.yellow}⚠ ${colors.reset}All errors will be treated as new.`);
+      return [];
+    }
+  } catch (error) {
+    console.error(`${colors.red}✖ ${colors.reset}Error loading baseline: ${error.message}`);
+    process.exit(1);
+  }
+}
+
+// Helper to run TypeScript compiler and get structured output
+function getCurrentErrors() {
```

Comment:
Can we separate a unified type check logic to share between `ts_error_checker.js` and `update_ts_baseline.js`?